src/mesh/interpolation/hermite_spline_z.cxx

```diff
@@ -62,20 +62,23 @@ void ZHermiteSpline::calcWeights(const Field3D& delta_z, const std::string& regi
 
     // The integer part of zt_prime are the indices of the cell
     // containing the field line end-point
-    k_corner[i.ind] = static_cast<int>(floor(delta_z(x, y, z)));
+    int corner_zind = floor(delta_z(x, y, z));
 
     // t_z is the normalised coordinate \in [0,1) within the cell
     // calculated by taking the remainder of the floating point index
-    const BoutReal t_z = delta_z(x, y, z) - static_cast<BoutReal>(k_corner[i.ind]);
+    const BoutReal t_z = delta_z(x, y, z) - static_cast<BoutReal>(corner_zind);
+
+    // make corner_zind be in the range 0<=corner_zind<nz
+    corner_zind = ((corner_zind % ncz) + ncz) % ncz;
```

Comment:
Move into the initialisation of `corner_zind` to make it `const`?

Reply:
No, because we need the 'uncorrected' version to calculate `t_z` in l.69. I'll add a comment.

Reply:
Ahh didn't spot that, thanks!